### PR TITLE
Russell-L210.resources

### PR DESCRIPTION
**Pull request by russellhz**
#707

Waiting on #810 for some units

### CODE REVIEW

File: R/zchunk_L210.resources.R [hunk 1]
```diff
@@ -90,364 +112,451 @@ module_energy_L210.resources_DISABLED <- function(command, ...) {
...
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code converts gdp using a conv_xxxx_xxxx_USD constant
-    # Use the `gdp_deflator(year, base_year)` function instead
+    # A. Output unit, price unit, market
+    L210.rsrc_info <- A10.rsrc_info %>%
+      # Repeat and add region to resource assumptions table
+      repeat_add_columns(select(GCAM_region_names, region)) %>%
+      # Remove traditional biomass from regions where it does not apply
```
Comment (bpbond) [inline R/zchunk_L210.resources.R]: Where does it not apply? Should this be clarified?

File: R/zchunk_L210.resources.R [hunk 2]
```diff
@@ -90,364 +112,451 @@ module_energy_L210.resources_DISABLED <- function(command, ...) {
...
+    # L210.GrdRenewRsrcCurves_geo: graded supply curves of geothermal (hydrothermal) resources
+    L210.GrdRenewRsrcCurves_geo <- L116.RsrcCurves_EJ_R_geo %>%
+      # Add region name
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      mutate(available = round(available, energy.DIGITS_MAX_SUB_RESOURCE)) %>%
+      select(region, renewresource = resource, sub.renewable.resource = subresource, grade, available, extractioncost)
+
+    # L210.GrdRenewRsrcMax_geo: default max sub resource of geothermal (hydrothermal) resources
+    # The old code is "subset( L210.GrdRenewRsrcCurves_geo, grade = unique( grade )[1] )"
+    # It appears they meant to filter to "grade 1" only, however there is only one equals sign, so no subsetting occurs
```
Comment (bpbond) [inline R/zchunk_L210.resources.R]: 👍

File: R/zchunk_batch_resources_SSP_xml.R [hunk 3]
```diff
@@ -1,37 +1,54 @@
...
 #' @param ... other optional parameters, depending on command
 #' @return Depends on \code{command}: either a vector of required inputs,
 #' a vector of output names, or (if \code{command} is "MAKE") all
-#' the generated outputs: \code{resources_SSP5.xml}. The corresponding file in the
-#' original data system was \code{batch_resources_SSP5_xml.R} (energy XML).
-module_energy_batch_resources_SSP_xml_DISABLED <- function(command, ...) {
+#' the generated outputs: \code{resources_SSP1.xml}, \code{resources_SSP2.xml}, \code{resources_SSP3.xml}, \code{resources_SSP4.xml}, \code{resources_SSP5.xml}.
+module_energy_batch_resources_SSP_xml <- function(command, ...) {
+
+  SSP_nums <- seq(1,5)
```
Comment (bpbond) [inline R/zchunk_batch_resources_SSP_xml.R]: `1:5`?

File: R/zchunk_L210.resources.R [hunk 4]
```diff
@@ -56,28 +62,44 @@ module_energy_L210.resources_DISABLED <- function(command, ...) {
...
+
     all_data <- list(...)[[1]]
 
     # Load required inputs
     GCAM_region_names <- get_data(all_data, "common/GCAM_region_names")
     A_regions <- get_data(all_data, "energy/A_regions")
-    A10.rsrc_info <- get_data(all_data, "energy/A10.rsrc_info")
+    A10.rsrc_info <- get_data(all_data, "energy/A10.rsrc_info") %>%
+      gather(year, value, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.integer(year))
```
Comment (bpbond) [inline R/zchunk_L210.resources.R]: Replace these new lines with call to new `gather_year` function ( #821 ). This also applies to all `gather %>% mutate(year = as.integer(year))` sequences below.

**Comment by bpbond** [comment]
Waiting on @gokuliyer review.

**Comment by d3y419** [comment]
@mwisepnnl review please?

**Comment by gokuliyer** [comment]
@bpbond: I received an email saying that this is waiting for my review. But I presume that was an error?

**Comment by bpbond** [comment]
@gokuliyer Yeah, my mistake, sorry.

**Pull request merged**